weld/subsystem/src/main/java/org/jboss/as/weld/deployment/processors/WeldDeploymentProcessor.java

```diff
@@ -279,7 +279,9 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
         }
 
         final EarMetaData earConfig = deploymentUnit.getAttachment(org.jboss.as.ee.structure.Attachments.EAR_METADATA);
-        if (earConfig == null || !earConfig.getInitializeInOrder())  {
+        final List<String> startupBeanModules = deploymentUnit.getAttachment(org.jboss.as.ee.component.Attachments.DEPLOYMENT_STARTUP_MODULES);
+
```

Comment:
I noticed in other places you used `getAttachmentList` instead of `getAttachment`.  `getAttachmentList` returns non-null value so no need to check null.